pkg/v1/tkg/tkgctl/helpers.go

```diff
@@ -187,8 +188,13 @@ func CheckIfInputFileIsClusterClassBased(clusterConfigFile string) (bool, unstru
 	for i := range yamlObjects {
 		obj := yamlObjects[i]
 		if obj.GetKind() == constants.KindCluster {
-			isInputFileClusterClassBased = true
 			clusterObj = obj
+			class, exists, _ := unstructured.NestedString((&clusterObj).UnstructuredContent(), "spec", "topology", "class")
+			if exists && class != "" {
+				isInputFileClusterClassBased = true
+			} else {
+				return isInputFileClusterClassBased, clusterObj, errors.New(constants.ClusterResourceWithoutTopologyNotSupportedErrMsg)
```

Comment:
any reason why we need to return an error here? Seems like the caller would have the proper context to determined from the returned results to assess if it is an error or not.